src/App.js

```diff
@@ -2,15 +2,45 @@ import React from 'react';
 import './App.scss';
 
 class App extends React.Component {
-  addOne = () => {};
+  state = {
+    counter: 0,
+  }
+
+  addOne = () => {
+    this.setState(({ counter }) => ({
+      counter: counter + 1,
+    }));
+  };
 
-  add100 = () => {};
+  add100 = () => {
+    this.setState(({ counter }) => ({
+      counter: counter + 100,
+    }));
+  };
 
-  increase = () => {};
+  increase = () => {
+    this.setState(({ counter }) => {
+      if (!(counter % 5)) {
+        this.add100();
+      }
+
+      this.addOne();
+    });
+  };
```

Comment:
```suggestion
  increase = () => {
    if (!(counter % 5)) {
      this.add100();
    }

    this.addOne();
  };
```
No need in setState here. You call two functions that already call setState method inside them